receiver/prometheusremotewritereceiver/receiver_test.go

```diff
@@ -406,17 +421,213 @@ func TestTranslateV2(t *testing.T) {
+				attrs.PutStr("region", "us-central1")
+				attrs.PutStr("datacenter", "sdc")
+
+				sm := rm.ScopeMetrics().AppendEmpty()
+				sm.Scope().SetName("OpenTelemetry Collector")
+				sm.Scope().SetVersion("latest")
+
+				m := sm.Metrics().AppendEmpty()
+				m.SetName("normal_metric")
+				m.SetUnit("")
+				m.SetDescription("")
+
+				dp := m.SetEmptyGauge().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(1.0)
+				dp.SetTimestamp(pcommon.Timestamp(1 * int64(time.Millisecond)))
+				dp.Attributes().PutStr("d", "e")
+
+				return metrics
+			}(),
+		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			metrics, stats, err := prwReceiver.translateV2(ctx, tc.request)
+			// since we are using the interRequestCache to store values across requests, we need to clear it after each test, otherwise it will affect the next test
+			prwReceiver.interRequestCache = make(map[uint64]pmetric.ResourceMetrics)
+			metrics, _, err := prwReceiver.translateV2(ctx, tc.request)
 			if tc.expectError != "" {
 				assert.ErrorContains(t, err, tc.expectError)
 				return
 			}
 
 			assert.NoError(t, err)
 			assert.NoError(t, pmetrictest.CompareMetrics(tc.expectedMetrics, metrics))
-			assert.Equal(t, tc.expectedStats, stats)
 		})
 	}
 }
+
+type nonMutatingConsumer struct{}
+
+// Capabilities returns the base consumer capabilities.
+func (bc nonMutatingConsumer) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+type MockConsumer struct {
+	nonMutatingConsumer
+	mu         sync.Mutex
+	metrics    []pmetric.Metrics
+	dataPoints int
+}
+
+func (m *MockConsumer) ConsumeMetrics(_ context.Context, md pmetric.Metrics) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.metrics = append(m.metrics, md)
+	m.dataPoints += md.DataPointCount()
+	return nil
+}
+
+func TestTargetInfoWithMultipleRequests(t *testing.T) {
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig()
+	cfg.(*Config).Endpoint = "localhost:9090"
+
+	mockConsumer := new(MockConsumer)
+	prwReceiver, err := factory.CreateMetrics(context.Background(), receivertest.NewNopSettings(metadata.Type), cfg, mockConsumer)
+	assert.NoError(t, err)
+	assert.NotNil(t, prwReceiver)
+
+	host := componenttest.NewNopHost()
+	err = prwReceiver.Start(context.Background(), host)
+	assert.NoError(t, err)
+	defer func() {
+		assert.NoError(t, prwReceiver.Shutdown(context.Background()))
+	}()
+
+	// Define two different metrics, one is target_info and the other is a normal metric.
+	// In the end, we expect that we have 1 final metric. All labels from the target_info metric should be translated to resource attributes and
+	// the normal metric should be translated as usual.
+	requests := []*writev2.Request{
+		{
+			Symbols: []string{
+				"",
+				"job", "production/service_a", // 1, 2
+				"instance", "host1", // 3, 4
+				"machine_type", "n1-standard-1", // 5, 6
+				"cloud_provider", "gcp", // 7, 8
+				"region", "us-central1", // 9, 10
+				"__name__", "target_info", // 11, 12
```

Comment:
I have a feeling that our current implementation only works if `target_info` is sent before the metric it was supposed to complement. If that's the behavior, that's totally ok, but let's also document this requirement in our README.

If I'm incorrect, let's also add a test case where target_info is sent after the original metric!

Reply:
[c4e8771](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38812/commits/c4e8771c1c1693bfd7d4bd865bc17e19519f780f)

We have both approaches working properly. I added a new test case here to assure it